spec/specification/linter_spec.rb

```diff
@@ -305,6 +305,38 @@ def result_should_include(*values)
 
       #------------------#
 
+      it 'fails a test spec with `requires_app_host = false` and `app_host_name` set' do
+        @spec.test_specification = true
+        @spec.requires_app_host = false
+        @spec.app_host_name = 'BananaLib/App'
+        result_should_include('app_host_name', 'requires_app_host')
+      end
+
+      it 'passes a test spec with `requires_app_host = true` and `app_host_name` set' do
+        @spec.test_specification = true
+        @spec.requires_app_host = true
+        @spec.app_host_name = 'BananaLib/App'
+        @linter.lint
+        @linter.results.should.be.empty?
+      end
+
+      it 'fails a test spec requiring an app host from a pod that isn\'t required' do
+        @spec.test_specification = true
+        @spec.requires_app_host = true
+        @spec.app_host_name = 'Foo/App'
+        result_should_include('app_host_name', 'Foo')
+      end
+
+      it 'passes a test spec requiring an app host from a pod that is listed as a dependency' do
```

Comment:
this test doesnt actually assert that the linter passes, rather it asserts that there is an error